extensions/ql-vscode/test/unit-tests/model-editor/yaml.test.ts

```diff
@@ -980,6 +981,132 @@ describe("createDataExtensionYamlsForFrameworkMode", () => {
+            provenance: "df-generated",
+            signature:
+              "Volo.Abp.TestApp.MongoDb.HostTestAppDbContext#get_FifthDbContextDummyEntity()",
+            packageName: "Volo.Abp.TestApp.MongoDb",
+            typeName: "HostTestAppDbContext",
+            methodName: "get_FifthDbContextDummyEntity",
+            methodParameters: "()",
+          },
+        ],
+      "Volo.Abp.TestApp.MongoDB.CityRepository#FindByNameAsync(System.String)":
+        [
+          {
+            type: "neutral",
+            kind: "summary",
+            provenance: "df-generated",
+            signature:
+              "Volo.Abp.TestApp.MongoDB.CityRepository#FindByNameAsync(System.String)",
+            packageName: "Volo.Abp.TestApp.MongoDB",
+            typeName: "CityRepository",
+            methodName: "FindByNameAsync",
+            methodParameters: "(System.String)",
+          },
+        ],
+    };
+    const modelYaml = `extensions:
+  - addsTo:
+      pack: codeql/csharp-all
+      extensible: sourceModel
+    data: []
+
+  - addsTo:
+      pack: codeql/csharp-all
+      extensible: sinkModel
+    data:
+      - ["Volo.Abp.TestApp.MongoDb","HostTestAppDbContext",true,"get_FifthDbContextDummyEntity","()","","Argument[0]","sql","df-generated"]
+
+  - addsTo:
+      pack: codeql/csharp-all
+      extensible: summaryModel
+    data: []
+
+  - addsTo:
+      pack: codeql/csharp-all
+      extensible: neutralModel
+    data:
+      - ["Volo.Abp.TestApp.MongoDB","CityRepository","FindByNameAsync","(System.String)","summary","df-generated"]
+`;
+
+    it("creates the correct YAML files when there are existing modeled methods", () => {
+      const yaml = createDataExtensionYamlsForFrameworkMode(
+        QueryLanguage.CSharp,
+        methods,
+        newModeledMethods,
+        {},
+      );
+
+      expect(yaml).toEqual({
+        "models/Volo.Abp.TestApp.MongoDB.model.yml": modelYaml,
+      });
+    });
+
+    it("creates the correct YAML files when there are existing modeled methods", () => {
+      const yaml = createDataExtensionYamlsForFrameworkMode(
+        QueryLanguage.CSharp,
+        methods,
+        newModeledMethods,
+        {
+          "models/Volo.Abp.TestApp.mongodb.model.yml": {
+            "Volo.Abp.TestApp.MongoDB.CityRepository#FindByNameAsync(System.String)":
+              [
+                {
+                  type: "neutral",
+                  kind: "summary",
+                  provenance: "manual",
+                  signature:
+                    "Volo.Abp.TestApp.MongoDB.CityRepository#FindByNameAsync(System.String)",
+                  packageName: "Volo.Abp.TestApp.MongoDB",
+                  typeName: "CityRepository",
+                  methodName: "FindByNameAsync",
+                  methodParameters: "(System.String)",
+                },
+              ],
+          },
+        },
+      );
+
+      expect(yaml).toEqual({
+        "models/Volo.Abp.TestApp.mongodb.model.yml": modelYaml,
```

Comment:
Why does this filename contain uppercase characters? I can see it has `mongodb` instead of `MongoDB` but I'm confused by the rest of the filename.

Reply:
This is based on the input filename on line 1086, this tests that it uses the same capitalization as an existing file. The actual capitalization of the file is arbitrary.

Reply:
Ah, thanks for explaining. That makes sense.